### PR TITLE
Change logic for AI castle development

### PR DESCRIPTION
**Pull request by ihhub**
- AI is able to build Black Dragons
- AI prioritizes dwellings of higher level to be built first
- AI doesn't build useless for him buildings
- AI builds some buildings only having certain amount of resources
- AI doesn't develop castle if no creatures inside and an enemy is nearby
- AI builds Well only on 7th day (no need to build before)
- AI builds tent into castle under more conditions

relates to #209

### CODE REVIEW

**Comment by ihhub** [comment]
Hi @andrey-starodubtsev would you mind to have a look into changes and put your thoughts if any? I've tested changes on my PC.

**Comment by andrey-starodubtsev** [comment]
Looks good!

One thing though: in `AICastleDevelopment` all monsters are bought as the last option. This means that AI will skip the last day if it squander all the money for building.

Should this statement be moved at the beginning of the `AICastleDevelopment` function?

Also I committed cosmetic formatting to comply with `.clang-format` and pass Travis CI. Feel free to drop this commit in case if you think it's too bulky.

**Comment by ihhub** [comment]
Hi @andrey-starodubtsev , this is an original logic for castle development. I know that it's weird and we need more sophisticated AI but for now we could stick to such logic. I fixed some obvious bugs in castle development logic, but of course we have a plenty of room for improvement.

File: src/fheroes2/ai/simple/ai_castle.cpp [hunk 1]
```diff
@@ -1,257 +1,202 @@
...
- *                                                                              *
- *   Part of the Free Heroes2 Engine:                                           *
- *   http://sourceforge.net/projects/fheroes2                                   *
- *                                                                              *
- *   Redistribution and use in source and binary forms, with or without         *
- *   modification, are permitted provided that the following conditions         *
- *   are met:                                                                   *
- *   - Redistributions may not be sold, nor may they be used in a               *
- *     commercial product or activity.                                          *
- *   - Redistributions of source code and/or in binary form must reproduce      *
```
Comment (andrey-starodubtsev) [inline src/fheroes2/ai/simple/ai_castle.cpp]: This PR violates this license statement, BTW.

Comment (ihhub) [reply src/fheroes2/ai/simple/ai_castle.cpp]: In my opinion I see no sense to store such text in each file as we mostly redo whole source code in it. Plus the date would be incorrect as we modify it 9 years later. We must mention original author and license in readme and license files instead. What do you think?

File: src/fheroes2/ai/simple/ai_castle.cpp [hunk 2]
```diff
@@ -1,257 +1,202 @@
...
+                BuildIfEnoughResources( castle, BUILD_MAGEGUILD2, GetResourceMultiplier( castle, 2, 3 ) ); // needed for Upg. Mausoleum
+                BuildIfEnoughResources( castle, BUILD_MAGEGUILD1, GetResourceMultiplier( castle, 2, 3 ) ); // needed for Mausoleum
+                BuildIfEnoughResources( castle, BUILD_THIEVESGUILD, GetResourceMultiplier( castle, 3, 4 ) ); // needed for Mansion
+            }
+            else if ( Race::WZRD == castle.GetRace() ) {
+                BuildIfEnoughResources( castle, BUILD_SPEC, GetResourceMultiplier( castle, 2, 3 ) ); // Library needed for Upg. Ivory Tower
+                BuildIfEnoughResources( castle, BUILD_MAGEGUILD1, GetResourceMultiplier( castle, 2, 3 ) ); // needed for Ivory Tower
+                BuildIfEnoughResources( castle, BUILD_WELL, GetResourceMultiplier( castle, 3, 4 ) ); // needed for Upg. Foundry
+            }
+            if ( world.LastDay() )
```
[the remark this answers is not in the document]
Comment (ihhub) [reply src/fheroes2/ai/simple/ai_castle.cpp]: Yes and no. If you look at line 103 you would notice that we build it if we have enough resources. I agree that logic is bad and has a lot of of ifs but it's better than previously.

File: src/fheroes2/ai/simple/ai_castle.cpp (hunk 1 above)
Comment (ihhub) [reply src/fheroes2/ai/simple/ai_castle.cpp]: I've return license comments for now until we decide what to do with them in correct way.

**Pull request merged**